core/src/test/java/org/infinispan/marshall/VersionAwareMarshallerTest.java

```diff
@@ -659,6 +712,75 @@ public void readExternal(ObjectInput in) throws IOException, ClassNotFoundExcept
       }
    }
 
+   public static class PojoExtended extends Pojo {
+      public PojoExtended() {}
+
+      public PojoExtended(int i, boolean b) {
+         super(i, b);
+      }
+   }
+
+   public static class PojoWithExternalizer extends Pojo {
+
+      public PojoWithExternalizer(int i, boolean b) {
+         super(i, b);
+      }
+
+      public static class Externalizer implements AdvancedExternalizer<PojoWithExternalizer> {
+         @Override
+         public Set<Class<? extends PojoWithExternalizer>> getTypeClasses() {
+            return Util.asSet(PojoWithExternalizer.class);
+         }
+
+         @Override
+         public Integer getId() {
+            return 1234;
+         }
+
+         @Override
+         public void writeObject(ObjectOutput output, PojoWithExternalizer object) throws IOException {
+            output.writeInt(object.i);
+            output.writeBoolean(object.b);
+         }
+
+         @Override
+         public PojoWithExternalizer readObject(ObjectInput input) throws IOException, ClassNotFoundException {
+            return new PojoWithExternalizer(input.readInt(), input.readBoolean());
+         }
+      }
+   }
+
+   public static class PojoWithMultiExternalizer extends Pojo {
+
+      public PojoWithMultiExternalizer(int i, boolean b) {
+         super(i, b);
+      }
+
+      public static class Externalizer implements AdvancedExternalizer<Object> {
+         @Override
+         public Set<Class<?>> getTypeClasses() {
+            return Util.asSet(PojoWithMultiExternalizer.class, Thread.class);
```

Comment:
`Thread.class`?

I think what you should be testing here is that a normal `Externalizer` (not advanced one, but annotated) work as expected.